offsets_store.go

```diff
@@ -469,7 +470,10 @@ func (storage *OffsetStorage) evaluateGroup(cluster string, group string, result
 
 			// Rule 4 - Offsets haven't been committed in a while
 			if ((time.Now().Unix() * 1000) - lastOffset.Timestamp) > (lastOffset.Timestamp - firstOffset.Timestamp) {
-				status.Status = protocol.StatusError
+				// Set consumer to error if there is also lag on partition
+				if lastOffset.Lag > 0 {
```

Comment:
This does not do what you think it does. `lastOffset.Lag` is only calculated at the time that consumer commits the offset. If the broker offset moves forward after this, the Lag value is not recalculated. This is the reason for the artificial offset commits that attempt to catch this case (where the topic is slow).

The check that you have added here means that the partition will never go into the error state if at the time the consumer stopped it was caught up.